### PR TITLE
feat(j-s): defentant party screen

### PR DESCRIPTION
**Pull request by Arnarkari93**
[Asana - Malflytjendur](https://app.asana.com/0/1199153462262248/1202861834321041/f)

## What

- Move `ProsecutorSelection` to components to be able to use also for indictments
- Move input for defender out of `DefenderInfo` to -> `DefenderInput`
- Add `ProsecutorAndDefender` screen to court indictments
- Add `defendantWaivesRightToCounsel ` field

## Screenshots / Gifs
Screenshot 2022-09-14 at 16 15 12

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: apps/judicial-system/api/src/app/modules/case/dto/updateCase.input.ts [hunk 1]
```diff
@@ -276,4 +276,8 @@ export class UpdateCaseInput implements UpdateCase {
   @Allow()
   @Field(() => String, { nullable: true })
   readonly subpoenaType?: SubpoenaType
+
+  @Allow()
+  @Field({ nullable: true })
+  readonly defendantRefusesHavingDefender?: boolean
```
Comment (oddsson) [inline apps/judicial-system/api/src/app/modules/case/dto/updateCase.input.ts:282]: I think a more accurate name for this is something like `defendantWaivesRightToCounsel`. What do you think @gudjong

File: apps/judicial-system/web/src/components/DefenderInfo/DefenderInfo.tsx [hunk 2]
```diff
@@ -1,22 +1,14 @@
-import React, { useContext, useState, useCallback } from 'react'
-import InputMask from 'react-input-mask'
+import React, { useContext, useState } from 'react'
```
Comment (oddsson) [inline apps/judicial-system/web/src/components/DefenderInfo/DefenderInfo.tsx:1]: There are a couple of unused imports in this file. Please remove them.

File: apps/judicial-system/web/src/components/DefenderInfo/DefenderInput.tsx [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+              event.target.value,
+              ['email-format'],
+              workingCase,
+              setWorkingCase,
+              emailErrorMessage,
+              setEmailErrorMessage,
+            )
+          }
+          onBlur={(event) =>
+            validateAndSendToServer(
```
Comment (oddsson) [inline apps/judicial-system/web/src/components/DefenderInfo/DefenderInput.tsx:129]: I thought we had removed this validateAndSendToServer 🤔 

Please consider using the `validateAndSendToServer` we created in the useCase hook.

File: apps/judicial-system/web/src/components/ProsecutorSelection/ProsecutorSelection.tsx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+        isIndictmentCase: isIndictmentCase(workingCase.type),
+      })}
+      placeholder={formatMessage(strings.placeholder, {
+        isIndictmentCase: isIndictmentCase(workingCase.type),
+      })}
+      value={selectedProsecutor}
+      options={availableProsecutors}
+      onChange={(value: ValueType<Option>) => {
+        const id = (value as ReactSelectOption).value
+        if (id && typeof id === 'string') {
```
Comment (oddsson) [inline apps/judicial-system/web/src/components/ProsecutorSelection/ProsecutorSelection.tsx:68]: Why are we checking the type of the id here?

**Review by oddsson — CHANGES_REQUESTED** [review]
Couple of questions

File: apps/judicial-system/api/src/app/modules/case/dto/updateCase.input.ts (hunk 1 above)
Comment (gudjong) [reply apps/judicial-system/api/src/app/modules/case/dto/updateCase.input.ts:282]: Agree @oddsson

Comment (Arnarkari93) [reply apps/judicial-system/api/src/app/modules/case/dto/updateCase.input.ts:282]: Perfect

File: apps/judicial-system/web/src/components/ProsecutorSelection/ProsecutorSelection.tsx (hunk 4 above)
Comment (Arnarkari93) [reply apps/judicial-system/web/src/components/ProsecutorSelection/ProsecutorSelection.tsx:68]: Select component typing in island-ui is pretty bad. `value` is typed `number | string` even though it is always a number in this case. This check is only shut up typescript.

**Review by oddsson — APPROVED** [review]

**Pull request merged**